pkg/operator/controller/controller_lb.go

```diff
@@ -118,20 +118,31 @@ func (r *reconciler) finalizeLoadBalancerService(ci *operatorv1.IngressControlle
 	if service == nil {
 		return nil
 	}
-	records, err := desiredDNSRecords(ci, service, dnsConfig)
-	if err != nil {
-		return err
-	}
-	dnsErrors := []error{}
-	for _, record := range records {
-		if err := r.DNSManager.Delete(record); err != nil {
-			dnsErrors = append(dnsErrors, fmt.Errorf("failed to delete DNS record %v for ingress %s/%s: %v", record, ci.Namespace, ci.Name, err))
-		} else {
-			log.Info("deleted DNS record for ingress", "namespace", ci.Namespace, "name", ci.Name, "record", record)
+	// If no load balancer has been provisioned, we cannot have published
+	// DNS records for it.
```

Comment:
Nit: this isn't necessarily true (hence the TODO): if no load balancer is _currently_ reported on status_ it _probably_ means we haven't published DNS

Reply:
Thanks! Updated comment.